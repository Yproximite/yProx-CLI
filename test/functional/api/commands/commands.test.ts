import { createFakeEnv } from '../../fake-env';

describe('api: commands', () => {
  it('should have registered commands', async () => {
    const { api, cleanup } = await createFakeEnv();

    expect(api.commands.build).toBeDefined();
    expect(api.commands.lint).toBeDefined();
    expect(api.commands).toMatchSnapshot();

    await cleanup();
  });

  describe('executeCommand', () => {
    it('should throw an error if no command passed', async () => {
      const { api, cleanup } = await createFakeEnv();

      expect(() => {
        // @ts-ignore
        api.executeCommand();
      }).toThrow(new Error('You must specify a command to run.'));

      await cleanup();
    });

    it('should throw an error if a command is not registered', async () => {
      const { api, cleanup } = await createFakeEnv();

      expect(() => {
        api.executeCommand('foobar');
      }).toThrow(new Error('Command "foobar" does not exist.'));

      await cleanup();
    });

    it('should execute the correct command', async () => {
      const { api, cleanup } = await createFakeEnv();

      api.commands.build.fn = jest.fn();
      api.executeCommand('build', { lint: true });
      expect(api.commands.build.fn).toHaveBeenCalledWith({ lint: true });

      api.commands.lint.fn = jest.fn();
      api.executeCommand('lint');
      expect(api.commands.lint.fn).toHaveBeenCalledWith({});

      await cleanup();
    });
  });
});
