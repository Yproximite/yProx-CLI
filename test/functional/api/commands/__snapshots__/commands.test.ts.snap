// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api: commands should have registered commands 1`] = `
Object {
  "build": Object {
    "fn": [Function],
    "name": "build",
    "opts": Object {
      "description": "build files",
      "options": Object {
        "--filter:<filterName> <value>": "apply a filter on file entries",
        "--help": "display help",
        "--lint": "lint before build, if lint fails, files will not be build",
        "--mode": "specify env mode (default: development)",
        "--watch": "enable watch mode",
      },
      "usage": "yprox-cli build [options]",
    },
  },
  "lint": Object {
    "fn": [Function],
    "name": "lint",
    "opts": Object {
      "description": "lint source files",
      "options": Object {
        "--filter:<filterName> <value>": "apply a filter on file entries",
        "--fix": "automatically fix lint errors",
        "--help": "display help",
        "--max-warnings": "number of warnings to trigger nonzero exit code - default: -1",
        "--mode": "specify env mode (default: development)",
      },
      "usage": "yprox-cli lint [options]",
    },
  },
}
`;
